Sources/Public/BacktraceBreadcrumb.swift

```diff
@@ -0,0 +1,150 @@
+    @objc func addBreadcrumb(_ message: String,
+                        level: BacktraceBreadcrumbLevel) -> Bool
+    @objc func addBreadcrumb(_ message: String,
+                        type: BacktraceBreadcrumbType) -> Bool
+}
+
+@objc public class BacktraceBreadcrumb: NSObject {
+    
+    private var enabledBreadcrumbTypes = [BacktraceBreadcrumbType]()
+        
+    private static let breadcrumbLogFileName = "bt-breadcrumbs-0";
+
+    public static var DEFAULT_MAX_LOG_SIZE_BYTES = 64000;
+        
+    private var breadcrumbsLogManager: BacktraceBreadcrumbsLogManager?
+    
+    private func breadcrumbLogPath() throws -> String {
+        var fileURL = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+        fileURL.appendPathComponent(BacktraceBreadcrumb.breadcrumbLogFileName)
+        return fileURL.path
+    }
+    
+    public func enableBreadCrumbs(_ breadCrumbTypes: [BacktraceBreadcrumbType] = BacktraceBreadcrumbType.all , maxLogSize: Int = DEFAULT_MAX_LOG_SIZE_BYTES) {
+        self.enabledBreadcrumbTypes = breadCrumbTypes
+        
+        do {
+            let fileURLPath = try breadcrumbLogPath()
+            breadcrumbsLogManager = BacktraceBreadcrumbsLogManager(fileURLPath, maxQueueFileSizeBytes: BacktraceBreadcrumb.DEFAULT_MAX_LOG_SIZE_BYTES)
+            
+        } catch {
+            BacktraceLogger.warning("\(error.localizedDescription) \nWhen enable breadcrumbs")
+        }
+    }
+    
+    public func disableBreadCrumbs() {
+        enabledBreadcrumbTypes.removeAll()
+    }
+    
+    public func addBreadcrumb(_ message: String,
+                              attributes:[String:Any]? = nil,
+                              type: BacktraceBreadcrumbType = BacktraceBreadcrumbType.manual,
+                              level: BacktraceBreadcrumbLevel = BacktraceBreadcrumbLevel.info) -> Bool {
+        if  let breadcrumbsLogManager = breadcrumbsLogManager, isBreadcrumbsEnabled {
+            return breadcrumbsLogManager.addBreadcrumb(message, attributes: attributes, type: type, level: level)
+        }
+        return false
+    }
+    
+    private var isBreadcrumbsEnabled: Bool {
+        return enabledBreadcrumbTypes.count > 0
+    }
+    
+    private var getCurrentBreadcrumbId : Int? {
+        return breadcrumbsLogManager?.getCurrentBreadcrumbId
+    }
+    
+    public func processReportBreadcrumbs(_ report: inout BacktraceReport) {
+        if !isBreadcrumbsEnabled {
+            return
+        }
+        guard let lastBreadcrumbId = getCurrentBreadcrumbId else {
+            return
+        }
+        do {
+            let fileURLPath = try breadcrumbLogPath()
+            report.attachmentPaths.append(fileURLPath)
+            report.attributes["breadcrumbs.lastId"] = lastBreadcrumbId
+        } catch {
+            BacktraceLogger.warning("\(error.localizedDescription) \nWhen process breadcrumbs report")
```

Comment:
Typo `process` should be `processing`

Reply:
Changes has been done and code committed